eth/protocols/eth/receipt.go

```diff
@@ -421,10 +440,14 @@ func blockReceiptsToNetwork69(blockReceipts, blockBody rlp.RawValue) ([]byte, er
 	)
 	outer := enc.List()
 	for i := 0; it.Next(); i++ {
-		txType, _ := nextTxType()
 		content, _, _ := rlp.SplitList(it.Value())
 		receiptList := enc.List()
-		enc.WriteUint64(uint64(txType))
+		if isStateSyncReceipt(i) {
+			enc.WriteUint64(uint64(0)) // TxType is always 0 for state-sync transactions
```

Comment:
It seems incompatible with [PIP-74](https://github.com/0xPolygon/Polygon-Improvement-Proposals/pull/509/files#diff-e65b1d8c93a5833fe6e4d0823528eb7664568c550ab1e9d35d1af0f713685cc1R39). I suppose this will change after a HF?

Reply:
Yes, based on whatever tx type we decide to have for state-sync txs, this will have a hard fork check.